### PR TITLE
Add 'inline' option for Button collection

### PR DESCRIPTION
**Pull request by Cosmin-Daniel**

### CODE REVIEW

File: tests/src/ViewModel/ButtonActionTest.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        $this->expectException(InvalidArgumentException::class);
+
+        Button::action('text', true, 'foo');
+    }
+
+    public function viewModelProvider() : array
+    {
+        return [
+            'basic' => [Button::action('Button action')],
+            'citation' => [Button::action('Button action (citation)', true, 'citation')],
```
Comment (smoqadam) [inline tests/src/ViewModel/ButtonActionTest.php:39]: Isn't it better to use constants that have been introdeced in Button class? 

something like this: 

```php
'citation' => [Button::action('Button action (citation)', true, Button::ACTION_VARIANT_CITATION)],
```

Comment (Cosmin-Daniel) [reply tests/src/ViewModel/ButtonActionTest.php]: You're right. I've changed them to constants.

File: tests/src/ViewModel/ButtonCollectionTest.php [hunk 2]
```diff
@@ -24,19 +24,37 @@ public function it_has_data()
...
+        $this->assertSameWithoutOrder($data, $blockAction->toArray());
     }
 
     public function viewModelProvider() : array
     {
         return [
             'minimum' => [new ButtonCollection([Button::link('text', 'path')])],
-            'complete' => [new ButtonCollection([Button::link('text', 'path')], true, true)],
+            'inline' => [new ButtonCollection([Button::action('text')], true)],
+            'centered and compact' => [new ButtonCollection([Button::link('text', 'path')], false, true, true)],
```
Comment (nlisgo) [inline tests/src/ViewModel/ButtonCollectionTest.php:57]: Reintroduce complete with all bool's set to true.

File: src/ViewModel/Button.php [hunk 3]
```diff
@@ -124,6 +129,31 @@ public static function link(
...
+        if ($variant) {
+            Assertion::choice($variant,
+                [
+                    self::ACTION_VARIANT_CITATION,
+                    self::ACTION_VARIANT_COMMENT,
+                    self::ACTION_VARIANT_DOWNLOAD,
+                    self::ACTION_VARIANT_SHARE
+                ]
+            );
+            $button->classes .= ' button--action icon icon-' . $variant;
```
[the remark this answers is not in the document]
Comment (Cosmin-Daniel) [reply src/ViewModel/Button.php]: Added 'button--action' as default class for action buttons.

File: tests/src/ViewModel/ButtonCollectionTest.php (hunk 2 above)
Comment (Cosmin-Daniel) [reply tests/src/ViewModel/ButtonCollectionTest.php:57]: 'complete' is now added with all values set to true, together with 'minimum', 'inline' and 'centered and compact'.

**Comment by nlisgo** [comment]
@Cosmin-Daniel can we have a separate PR just for the change to button-collection to preserve existing behaviour and this can be a followup?

**Comment by Cosmin-Daniel** [comment]
> @Cosmin-Daniel can we have a separate PR just for the change to button-collection to preserve existing behaviour and this can be a followup?
> 
> Without this we will be blocked from updating styles on journal.

Ok. I will keep this one for the button-collection and create another as a follow-up with the action buttons changes.

**Pull request closed without merging**